tools/integration_tests/run_tests_mounted_directory.sh

```diff
@@ -304,3 +304,122 @@ echo "logging:
+  stat-cache-max-size-mb: 4
+  ttl-secs: $cache_ttl
+  type-cache-max-size-mb: 32
+cache-dir: /tmp/cache-dir" > /tmp/gcsfuse_config.yaml
+}
+
+# cache-file-for-range-read:false read-cache test.
+read_cache_test_setup 50 false 3600
+test_cases=(
+  "TestCacheFileForRangeReadFalseTest/TestRangeReadsWithCacheMiss"
+  "TestCacheFileForRangeReadFalseTest/TestConcurrentReads_ReadIsTreatedNonSequentialAfterFileIsRemovedFromCache"
+)
+for test_case in "${test_cases[@]}"; do
+  gcsfuse --config-file=/tmp/gcsfuse_config.yaml "$TEST_BUCKET_NAME" "$MOUNT_DIR" > /dev/null
+  GODEBUG=asyncpreemptoff=1 go test ./tools/integration_tests/read_cache/... -p 1 --integrationTest -v --mountedDirectory="$MOUNT_DIR" --testbucket="$TEST_BUCKET_NAME" -run "$test_case"
+  sudo umount "$MOUNT_DIR"
+  cleanup_test_environment
+done
+
+# cache-file-for-range-read:true read-cache test.
+read_cache_test_setup 50 true 3600
+test_case="TestCacheFileForRangeReadTrueTest/TestRangeReadsWithCacheHit"
+gcsfuse --config-file=/tmp/gcsfuse_config.yaml "$TEST_BUCKET_NAME" "$MOUNT_DIR" > /dev/null
+GODEBUG=asyncpreemptoff=1 go test ./tools/integration_tests/read_cache/... -p 1 --integrationTest -v --mountedDirectory="$MOUNT_DIR" --testbucket="$TEST_BUCKET_NAME" -run "$test_case"
+sudo umount "$MOUNT_DIR"
+cleanup_test_environment
+
+
+# Disabled cache ttl read-cache test.
+read_cache_test_setup 9 false 0
+test_case="TestDisabledCacheTTLTest/TestReadAfterObjectUpdateIsCacheMiss"
+gcsfuse --config-file=/tmp/gcsfuse_config.yaml "$TEST_BUCKET_NAME" "$MOUNT_DIR" > /dev/null
+GODEBUG=asyncpreemptoff=1 go test ./tools/integration_tests/read_cache/... -p 1 --integrationTest -v --mountedDirectory="$MOUNT_DIR" --testbucket="$TEST_BUCKET_NAME" -run "$test_case"
+sudo umount "$MOUNT_DIR"
+cleanup_test_environment
+
+# local modification test.
+read_cache_test_setup 9 false 3600
+test_case="TestLocalModificationTest/TestReadAfterLocalGCSFuseWriteIsCacheMiss"
+gcsfuse --config-file=/tmp/gcsfuse_config.yaml "$TEST_BUCKET_NAME" "$MOUNT_DIR" > /dev/null
+GODEBUG=asyncpreemptoff=1 go test ./tools/integration_tests/read_cache/... -p 1 --integrationTest -v --mountedDirectory="$MOUNT_DIR" --testbucket="$TEST_BUCKET_NAME" -run "$test_case"
+sudo umount "$MOUNT_DIR"
+cleanup_test_environment
+
+# range read read-cache test.
+read_cache_test_setup 500 false 3600
+test_cases=(
+  "TestRangeReadTest/TestRangeReadsWithinReadChunkSize"
+  "TestRangeReadTest/TestRangeReadsBeyondReadChunkSizeWithChunkDownloaded"
+  "TestRangeReadTest/TestRangeReadsBeyondReadChunkSizeWithoutChunkDownloaded"
+)
+for test_case in "${test_cases[@]}"; do
+  gcsfuse --config-file=/tmp/gcsfuse_config.yaml "$TEST_BUCKET_NAME" "$MOUNT_DIR" > /dev/null
+  GODEBUG=asyncpreemptoff=1 go test ./tools/integration_tests/read_cache/... -p 1 --integrationTest -v --mountedDirectory="$MOUNT_DIR" --testbucket="$TEST_BUCKET_NAME" -run "$test_case"
+  sudo umount "$MOUNT_DIR"
+  cleanup_test_environment
+done
+
+# Read only mount read-cache test.
```

Comment:
Here also, we have only considered enable_range_read as false, should we replicate it for true?